format/FormatNexusEigerDLS.py

```diff
@@ -0,0 +1,90 @@
+from __future__ import absolute_import, division, print_function
+
+import ast
+
+import h5py
+
+from dxtbx.format.FormatNexus import FormatNexus
+
+
+def get_count_limit_from_meta(meta_file_name):
+    with h5py.File(meta_file_name, "r") as f:
+
+        config = f["/config"][()]
+        config_data = ast.literal_eval(config.decode("utf-8"))
+
+    return config_data["countrate_correction_count_cutoff"]
+
+
+def get_bit_depth_from_meta(meta_file_name):
+    with h5py.File(meta_file_name, "r") as f:
+
+        config = f["/config"][()]
+        config_data = ast.literal_eval(config.decode("utf-8"))
+
+    return config_data["bit_depth_image"]
+
+
+class FormatNexusEigerDLS(FormatNexus):
+    @staticmethod
+    def understand(image_file):
+        # Get the file handle
+        with h5py.File(image_file, "r") as handle:
+            name = FormatNexusEigerDLS.get_instrument_name(handle)
+            if name is None or name.lower() not in (b"i03", b"i04", b"vmxi"):
+                return False
+
+        return True
+
+    def __init__(self, image_file, **kwargs):
+        """Initialise the image structure from the given file."""
+
+        super(FormatNexusEigerDLS, self).__init__(image_file, **kwargs)
+        try:
+            if self._image_file.endswith("_master.h5"):
+                meta = self._image_file.replace("_master.h5", "_meta.h5")
+            elif self._image_file.endswith(".nxs"):
+                meta = self._image_file.replace(".nxs", "_meta.h5")
+            self._bit_depth_image = get_bit_depth_from_meta(meta)
+        except Exception:
+            self._bit_depth_image = 0
+
+    def get_detector(self, index=None):
+        # workaround for https://jira.diamond.ac.uk/browse/I03-365
+        # read the count limit from the meta file - if anything goes
+        # wrong, do nothing
+
+        detector = self._detector()
+
+        try:
+            if self._image_file.endswith("_master.h5"):
+                meta = self._image_file.replace("_master.h5", "_meta.h5")
+            elif self._image_file.endswith(".nxs"):
+                meta = self._image_file.replace(".nxs", "_meta.h5")
+            limit = get_count_limit_from_meta(meta)
+
+            assert limit > 0
+
+            for panel in detector:
+                trusted = panel.get_trusted_range()
+                panel.set_trusted_range((trusted[0], limit))
+
+        except Exception:
+            pass
```

Comment:
Does all of this code need to be within the try/except? It should probably be something like:
```suggestion
        try:
            limit = get_count_limit_from_meta(self._meta)
            assert limit > 0
        else:
            for panel in detector:
                trusted = panel.get_trusted_range()
                panel.set_trusted_range((trusted[0], limit))
        except Exception:
            pass
```

Reply:
Glad I did not commit suggestion - the correct structure is:

```
try:
  this
except:
  that
else:
  finish
```

🙂

However done in 74053d1687bfe58802131d13182512da83422687

☝️  this does actually improve readability and makes more sense as syntax, so TIL